### DISCUSSION TITLE
Add connection pooling for dagster_postgres lib

### DISCUSSION THREAD

**Discussion by Lukashab** [dagster-io/dagster#20561] [Ideas]
Recently, I ran into an issue with dagster_postgres library #19222 . When I use postgres as a storage for my dagster architecture, I sometimes get "too many clients" error, especially when multiple jobs are running in parallel and there are concurrency slots set up for some ops. 
I was wondering why there is no pooling mechanism when connecting to postgres instance. 
For example here: https://github.com/dagster-io/dagster/blob/d3720b489d04cb634eb14b5a099b45f5ba0002b9/python_modules/libraries/dagster-postgres/dagster_postgres/event_log/event_log.py#L91 

Is there any reason behind this logic? Because for dagster instance with multiple shcedules and jobs running, this can cause failures very quickly and there is simply no other way how to avoid it than setting external pooling mechanism or extending the connection limit of the postgres instance.
Thank you

**Comment by the4thamigo-uk** [comment]
I suppose by 'external pooling mechanism' you mean https://www.pgbouncer.org/ or similar?

**Reply by Lukashab** [reply]
Yes exactly. Pgbouncer is what I meant. But, as I said,  i would rather have possibility to pool the connections on the dagster's side as well.

**Reply by the4thamigo-uk** [reply]
In also would love this feature as we are struggling with similar issues.
